src/cbdata.cc

```diff
@@ -297,12 +292,7 @@ cbdataRealFree(cbdata *c, const char *file, const int line)
 void *
 cbdataInternalFree(void *p, const char *file, int line)
 {
-    cbdata *c;
-#if WITH_VALGRIND
-    c = cbdata_htable.at(p);
-#else
-    c = (cbdata *) (((char *) p) - cbdata::Offset);
-#endif
+    auto *c = cbdata::FromUserData(p);
```

Comment:
Some of these new/changed declarations can probably be tightened up because the value of `c` never changes:

```suggestion
    const auto c = cbdata::FromUserData(p);
```

Reply:
Most of them need to be non-const for addHistory() use when USE_CBDATA_DEBUG is enabled. The one which does not is updated in 207b801.

Reply:
The proposed `const auto c = ...` change does not affect addHistory(). Again, the value of `c` never changes. You are probably thinking about the value of `*c` which the proposed expression does not make constant.